packages/http-client-python/emitter/src/emitter.ts

```diff
@@ -78,50 +80,113 @@ export async function $onEmit(context: EmitContext<PythonEmitterOptions>) {
+      const keyValuePairs = Object.entries(resolvedOptions["packaging-files-config"]).map(
+        ([key, value]) => {
+          return `${key}:${value}`;
+        },
+      );
+      commandArgs["packaging-files-config"] = keyValuePairs.join("|");
+      resolvedOptions["packaging-files-config"] = undefined;
+    }
+    if (
+      resolvedOptions["package-pprint-name"] !== undefined &&
+      !resolvedOptions["package-pprint-name"].startsWith('"')
+    ) {
+      resolvedOptions["package-pprint-name"] = `${resolvedOptions["package-pprint-name"]}`;
+    }
 
-  for (const [key, value] of Object.entries(resolvedOptions)) {
-    commandArgs.push(`--${key}=${value}`);
-  }
-  if (sdkContext.arm === true) {
-    commandArgs.push("--azure-arm=true");
-  }
-  if (resolvedOptions.flavor === "azure") {
-    commandArgs.push("--emit-cross-language-definition-file=true");
-  }
-  commandArgs.push("--from-typespec=true");
-  if (!program.compilerOptions.noEmit && !program.hasError()) {
-    execSync(commandArgs.join(" "));
+    for (const [key, value] of Object.entries(resolvedOptions)) {
+      commandArgs[key] = value;
+    }
+    if (sdkContext.arm === true) {
+      commandArgs["azure-arm"] = "true";
+    }
+    if (resolvedOptions.flavor === "azure") {
+      commandArgs["emit-cross-language-definition-file"] = "true";
+    }
+    commandArgs["from-typespec"] = "true";
+
+    if (!program.compilerOptions.noEmit && !program.hasError()) {
+      const outputFolder = path.relative(root, outputDir);
+      if (!fs.existsSync(outputFolder)) {
+        fs.mkdirSync(outputFolder, { recursive: true });
+      }
+      const pyodide = await loadPyodide({ indexURL: path.join(root, "node_modules", "pyodide") });
+      pyodide.FS.mount(pyodide.FS.filesystems.NODEFS, { root: "." }, ".");
+      await pyodide.loadPackage("setuptools");
+      await pyodide.loadPackage("tomli");
+      await pyodide.loadPackage("docutils");
+      await pyodide.loadPackage("micropip");
+      const micropip = pyodide.pyimport("micropip");
+      await micropip.install("emfs:generator/dist/pygen-0.1.0-py3-none-any.whl");
+      const yamlRelativePath = path.relative(root, yamlPath);
+      const globals = pyodide.toPy({ outputFolder, yamlRelativePath, commandArgs });
+      const python = `
+        async def main():
+          import warnings
+          with warnings.catch_warnings():
+            warnings.simplefilter("ignore", SyntaxWarning)
+            from pygen import m2r, preprocess, codegen, black
+
+          m2r.M2R(output_folder=outputFolder, cadl_file=yamlRelativePath, **commandArgs).process()
+          preprocess.PreProcessPlugin(output_folder=outputFolder, cadl_file=yamlRelativePath, **commandArgs).process()
+          codegen.CodeGenerator(output_folder=outputFolder, cadl_file=yamlRelativePath, **commandArgs).process()
+          black.BlackScriptPlugin(output_folder=outputFolder, **commandArgs).process()
+    
+        await main()
```

Comment:
is pyodide able to run a py file? i just wondering if two paths could be consolidated. previously, we used py file. here, we use inline code.